### DISCUSSION TITLE
MuPDF Integrated OCR Support

### DISCUSSION THREAD

**Discussion by JorjMcKie** [pymupdf/PyMuPDF#761] [Announcements]
MuPDF v1.18.0 contains support for integrated OCR via [Tesseract](https://github.com/tesseract-ocr/tesseract).
In principle, this is done by internally making a page image and then letting Tesseract analyze it. The recognized text is then handed on to the legacy MuPDF textpage functions.

I have not yet provided support for this in PyMuPDF. The reasons are similar as in the case of missing [PDF signing](https://github.com/pymupdf/PyMuPDF/discussions/757) feature:
* Tesseract and adequate recognition training data must be installed separately.
* MuPDF either supports an independent local Tesseract installation or an integrated Tesseract.
* For preparing wheels, this presents a significant challenge: obviously a decision between these alternatives must be made: wheels with or without Tesseract.
    - PyMuPDF installations which should use an **_independent Tesseract_** must be told, where to look for it - unclear yet how this may work.
    - **_Integrated Tesseract_** installations will be large and probably not well-trained, or at least the recognition training approach is unclear.

My current assessment is to aim for a PyMuPDF that **_also_** supports OCR, if it determines that Tesseract is installed. So the PyMuPDF user would have the option to independently install and train Tesseract if / when required.
PyMuPDF would have to appropriately react if OCR functions are used and Tesseract cannot be found.

Your comments are welcome1

**Comment by bserg66** [comment]
Good idea - call independent tesseract from PyMuPdf, with most suitable for PyMuPdf default settings.
I made OCR file wifh OCRMyPdf(tesseract) and got huge garbage after  PyMuPdf compression (see our discussions - Most terrible example).

**Comment by Joe5522** [comment]
Have you considered using Windows Media OCR? Just a thought.
